nexus/src/app/update.rs

```diff
@@ -81,17 +99,78 @@ impl super::Nexus {
     ) -> Result<TufRepoDescription, HttpError> {
         opctx.authorize(authz::Action::Read, &authz::FLEET).await?;
 
-        let _updates_config =
-            self.updates_config.as_ref().ok_or_else(|| {
-                Error::internal_error("updates system not initialized")
-            })?;
-
-        let tuf_repo_description = self
-            .db_datastore
+        self.db_datastore
             .tuf_repo_get_by_version(opctx, system_version.into())
             .await
-            .map_err(HttpError::from)?;
+            .map_err(HttpError::from)
+    }
+
+    pub(crate) async fn updates_add_trust_root(
+        &self,
+        opctx: &OpContext,
+        trust_root: serde_json::Value,
+    ) -> Result<TufTrustRoot, HttpError> {
+        opctx.authorize(authz::Action::Modify, &authz::FLEET).await?;
+
+        // Verify that this appears to be a valid TUF root role.
+        let parsed: Signed<Root> =
+            match serde_json::from_value(trust_root.clone()) {
+                Ok(parsed) => Ok(parsed),
+                Err(err) => Err(HttpError::for_bad_request(
+                    None,
+                    format!("invalid root role: {err}"),
+                )),
+            }?;
+        // Verify the root role is self-signed.
+        if let Err(err) = parsed.signed.verify_role(&parsed) {
+            return Err(HttpError::for_bad_request(
+                None,
+                format!("invalid root role: {err}"),
```

Comment:
Can we use `InlineErrorChain` here and at L121 to make sure we get information about any causes if `tough` errors have them (now or in the future)?

Reply:
This code is moving around with the wrapper struct commit, but this is going to result in repetitive error messages as tough currently includes cause messages in its Display strings. I can still do that if you want though.

(Idly wondering about a variant of `InlineErrorChain` that detects repeated causes and removes them...)

Reply:
Either seems okay right now.  In the limit I imagine we want to both fix `tough` (assuming we can) and fix this to use `InlineErrorChain`.  We could use `InlineErrorChain` here now, get the duplicates, and that'll be cleaned up when we fix `tough`.  Or we could file an issue to fix this when we pull in a newer `tough`.